File: resources/java_pojo_class_backup/trash/Amounts.java
package com.piccmaq.flutter_paypal_native.models.approvaldata.trash;


import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;


public class Amounts {

    @SerializedName("handlingFee")
    @Expose
    private Amount handlingFee;
    @SerializedName("insurance")
    @Expose
    private Amount insurance;
    @SerializedName("shippingAndHandling")
    @Expose
    private Amount shippingAndHandling;
    @SerializedName("shippingDiscount")
    @Expose
    private Amount shippingDiscount;
    @SerializedName("subtotal")
    @Expose
    private Amount subtotal;
    @SerializedName("tax")
    @Expose
    private Amount tax;
    @SerializedName("total")
    @Expose
    private String total;

    public Amount getHandlingFee() {
        return handlingFee;
    }

    public void setHandlingFee(Amount handlingFee) {
        this.handlingFee = handlingFee;
    }

    public Amount getInsurance() {
        return insurance;
    }

    public void setInsurance(Amount insurance) {
        this.insurance = insurance;
    }

    public Amount getShippingAndHandling() {
        return shippingAndHandling;
    }

    public void setShippingAndHandling(Amount shippingAndHandling) {
        this.shippingAndHandling = shippingAndHandling;
    }

    public Amount getShippingDiscount() {
        return shippingDiscount;
    }

    public void setShippingDiscount(Amount shippingDiscount) {
        this.shippingDiscount = shippingDiscount;
    }

    public Amount getSubtotal() {
        return subtotal;
    }

    public void setSubtotal(Amount subtotal) {
        this.subtotal = subtotal;
    }

    public Amount getTax() {
        return tax;
    }

    public void setTax(Amount tax) {
        this.tax = tax;
    }

    public String getTotal() {
        return total;
    }

    public void setTotal(String total) {
        this.total = total;
    }
}

File: lib/models/approval/amounts.dart
import 'package:flutter_paypal_native/models/approval/unit_price.dart';

class Amounts {
  UnitPrice? handlingFee;
  UnitPrice? insurance;
  UnitPrice? shippingAndHandling;
  UnitPrice? shippingDiscount;
  UnitPrice? subtotal;
  UnitPrice? tax;
  UnitPrice? total;

  Amounts(
      {this.handlingFee,
      this.insurance,
      this.shippingAndHandling,
      this.shippingDiscount,
      this.subtotal,
      this.tax,
      this.total});

  Amounts.fromJson(Map<String, dynamic> json) {
    handlingFee = json['handlingFee'] != null
        ? UnitPrice.fromJson(json['handlingFee'])
        : null;
    insurance = json['insurance'] != null
        ? UnitPrice.fromJson(json['insurance'])
        : null;
    shippingAndHandling = json['shippingAndHandling'] != null
        ? UnitPrice.fromJson(json['shippingAndHandling'])
        : null;
    shippingDiscount = json['shippingDiscount'] != null
        ? UnitPrice.fromJson(json['shippingDiscount'])
        : null;
    subtotal =
        json['subtotal'] != null ? UnitPrice.fromJson(json['subtotal']) : null;
    tax = json['tax'] != null ? UnitPrice.fromJson(json['tax']) : null;
    total = json['total'] != null ? UnitPrice.fromJson(json['total']) : null;
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = {};
    if (handlingFee != null) {
      data['handlingFee'] = handlingFee!.toJson();
    }
    if (insurance != null) {
      data['insurance'] = insurance!.toJson();
    }
    if (shippingAndHandling != null) {
      data['shippingAndHandling'] = shippingAndHandling!.toJson();
    }
    if (shippingDiscount != null) {
      data['shippingDiscount'] = shippingDiscount!.toJson();
    }
    if (subtotal != null) {
      data['subtotal'] = subtotal!.toJson();
    }
    if (tax != null) {
      data['tax'] = tax!.toJson();
    }
    if (total != null) {
      data['total'] = total!.toJson();
    }
    return data;
  }
}
